### PR TITLE
server/rotation.go: Fix key rotation with multiple dex instances.

### PR DESCRIPTION
**Pull request by rithujohn191**
This fixes https://github.com/coreos/dex/issues/976.

It also ensures that it displays "keys already rotated" as an info log message instead of an error message.

### CODE REVIEW

File: server/rotation.go [hunk 1]
```diff
@@ -15,6 +16,8 @@ import (
 	"github.com/coreos/dex/storage"
 )
 
+var errAlreadyRotated = errors.New("key already rotated by another instance of Dex")
```
Comment (ericchiang) [inline server/rotation.go]: nit:

s/key/keys/g
s/another instance of Dex/another server instance/g

**Comment by robszumski** [comment]
From the changes it looks like this was cosmetic not an actual bug? Is that true?

**Comment by ericchiang** [comment]
@robszumski returning `nil` instead of an error was causing the update to write an empty keys object to the backing storage. This was indeed a pretty bad bug if two instances of dex were attempting to rotate keys within a second of each other.

**Comment by rithujohn191** [comment]
@robszumski Adding to that its necessary to have more than one instance of Dex to reproduce this bug. Its not an issue if the user has only one replica of Dex

**Pull request merged**